Task 3/Task 3.2/Task 3.2/Task 3.2/DynamicArray.cs

```diff
@@ -0,0 +1,250 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Task_3._2
+{
+    public class DynamicArray<T> : IEnumerable<T>, ICloneable
+    {
+        private protected T[] _myArray { get; set; }
+
+        public int capacity { get; private set; } = 8;
+
+        public int length { get; private set; } = 0;
+
+        public DynamicArray()
+        {
+            _myArray = new T[capacity];
+        }
+
+        public DynamicArray(int value)
+        {
+            if (value > capacity)
+            {
+                while (capacity < value)
+                {
+                    capacity *= 2;
+                }
+            }
+
+            _myArray = new T[capacity];
+        }
+
+        public DynamicArray(ICollection<T> list)
+        {
+            if (list.Count > capacity)
+            {
+                while (capacity < list.Count)
+                {
+                    capacity *= 2;
+                }
+            }
+            _myArray = new T[capacity];
+            list.CopyTo(_myArray, 0);
+            length = list.Count;
+        }
+
+        public void Add(T item)
+        {
+            if (length + 1 <= capacity)
+            {
+                _myArray[length] = item;
+                length++;
+            }
+            else if (length + 1 > capacity)
+            {
+                T[] array = new T[capacity * 2];
+                for (int i = 0; i <= length; i++)
+                {
+                    if (i == length)
+                        array[i] = item;
+                    else
+                        array[i] = _myArray[i];
+                }
+                _myArray = array;
+                capacity = array.Length;
+                length++;
+            }
+        }
+
+        public void AddRange(ICollection<T> list)
+        {
+            while (capacity < length + list.Count)
+            {
+                capacity *= 2;
+            }
+            T[] array = new T[capacity];
+            for (int i = 0; i < length; i++)
+            {
+                array[i] = _myArray[i];
+            }
+            list.CopyTo(array, length);
+            _myArray = array;
+            length += list.Count;
+        }
+
+        public bool Remove(T item)
+        {
+            int index = Array.IndexOf(_myArray, item);
+            if (index == -1)
+            {
+                return false;
+            }
+            else
+            {
+                _myArray[index] = default(T);
+                for (int i = index; i <= length; i++)
+                {
+                    _myArray[i] = _myArray[i + 1];
+                    if (i == length)
+                    {
+                        _myArray[i] = default(T);
+                    }
+                }
```

Comment:
* `Array.Copy`
* you don't need to default last element since it's out of reach anyway (it's index is = length, out of range)